packages/compiler/src/core/types.ts

```diff
@@ -2334,6 +2335,8 @@ export interface LinterRuleDefinition<N extends string, DM extends DiagnosticMes
   url?: string;
   /** Messages that can be reported with the diagnostic. */
   messages: DM;
+  /** Whether this is an async rule. Default is false */
+  async?: boolean;
```

Comment:
can we make the type only show you can return promise for the exit even when this is true